### PR TITLE
ENT-3716: Match PAYG subscriptions by offering product name rather th…

### PR DESCRIPTION
**Pull request by dmckoske**
...rather than role.  This is similar to ENT-3610, which uses mostly unit testing to test it.

### CODE REVIEW

File: src/main/java/org/candlepin/subscriptions/files/ProductProfile.java [hunk 1]
```diff
@@ -92,29 +96,13 @@ public ProductProfile(String name, Set<SubscriptionWatchProduct> products, Granu
         setProducts(products);
     }
 
-    public String getName() {
```
Comment (awood) [inline src/main/java/org/candlepin/subscriptions/files/ProductProfile.java:95]: Why did these get stripped out?

**Review by awood — CHANGES_REQUESTED** [review]
Overall, it looks okay; I just have questions around why a lot of getters and setters got trimmed out of `ProductProfile`

File: src/main/java/org/candlepin/subscriptions/files/ProductProfile.java (hunk 1 above)
Comment (dmckoske) [reply src/main/java/org/candlepin/subscriptions/files/ProductProfile.java:95]: I replaced them with the Lombok annotations.

Comment (lindseyburnett) [reply src/main/java/org/candlepin/subscriptions/files/ProductProfile.java:95]: @awood, it looks like a previous PR made the lombok change but these weren't deleted as part of that

**Review by awood — APPROVED** [review]

**Pull request merged**